src/spatialref.jl

```diff
@@ -1,3 +1,85 @@
+"""
+Import format and run function `f` on the result.
+
+The name could be improved?
+"""
+importcrs(f::Function, sourceformat::GFT.GeoFormat) = begin
```

Comment:
Please write this as
```julia
function importcrs(f::Function, sourceformat::GFT.GeoFormat)
    obj = unsafe_importcrs(sourceformat)
    try f(obj) finally destroy(obj) end
end
```
The way to do it is to add `importcrs` to [this line](https://github.com/yeesian/ArchGDAL.jl/blob/fe9ee2a312eafac9d00f9f43100b0dd7f5a005e6/src/context.jl#L205). All the methods below it should be renamed from 
```
importcrs(x) = import*(GFT.val(x))
```
to
```
unsafe_importcrs(x) = unsafe_import*(GFT.val(x))
```
(Otherwise the spatialref object that is created will be destroyed twice. It will also make it easier to write a version that doesn't require you to wrap it in a do-block. If you're interested, see [this section of the docs](http://yeesian.com/ArchGDAL.jl/latest/memory/#Memory-Management-(Advanced)-1) for an explanation.)

Reply:
Oh right. You should document/comment that context code it's pretty hard to understand what you are doing there but important to how the package works.

Reply:
Done